examples/lang2/delegator/README.md

```diff
@@ -0,0 +1,29 @@
+# Delegator Smart Contract
+
+The delegator smart contract is our showcase for executing other smart contracts on-chain.
+
+It consists in total of 4 different smart contract:
+
+- Delegator (root): Delegates calls either to the Adder or Subber smart contract
+- Adder: Increases a value in the Accumulator smart contract
+- Subber: Decreases a value in the Accumulator smart contract
+- Accumulator: Owns a simple `i32` value that can be incremented or decremented
+
+In order to test this bundle of smart contracts you need to do the following:
+
+1. Compile all dependencies of the Delegator smart contract using the `./build-all.sh` script.
+   As usual you will receive their respective Wasm blobs in their respective `target` folders.
+1. Go to the delegator directory (root of the example) and run `cargo run -p abi-gen` in order
+   to generate the ABI file for the Delegator smart contract.
```

Comment:
Is there a reason why the ABI generation doesn't happen as part of `build-all.sh` and is a separate step instead?

Reply:
Yes: ABI generation is not as often required as building a contract but just when interfaces/signatures in it do change. Also since it requires a completely different kind of compilation, having the ABI generation step being in this script would significantly increase overall runtime of the script for near to no use.